dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunction.cs

```diff
@@ -287,16 +277,20 @@ internal KernelFunction(string name, string? pluginName, string description, IRe
                 // Quick check for cancellation after logging about function start but before doing any real work.
                 cancellationToken.ThrowIfCancellationRequested();
 
-                // Invoke pre-invocation filter. If it requests cancellation, throw.
-                var invokingContext = kernel.OnFunctionInvokingFilter(this, arguments);
-
-                if (invokingContext?.Cancel is true)
+                var invocationContext = await kernel.OnFunctionInvocationAsync(this, arguments, (context) =>
                 {
-                    throw new OperationCanceledException("A function filter requested cancellation before function invocation.");
-                }
+                    // Invoke the function and get its streaming enumerable.
+                    var enumerable = this.InvokeStreamingCoreAsync<TResult>(kernel, context.Arguments, cancellationToken);
 
-                // Invoke the function and get its streaming enumerator.
-                enumerator = this.InvokeStreamingCoreAsync<TResult>(kernel, arguments, cancellationToken).GetAsyncEnumerator(cancellationToken);
+                    // Update context with enumerable as result value.
+                    context.Result = new FunctionResult(this, enumerable);
```

Comment:
Kernel culture can be added to the function result as well.

Reply:
Good point, updated.